include/ghex/transport_layer/util/barrier.hpp

```diff
@@ -70,7 +70,8 @@ namespace gridtools {
                 template <typename TLCommunicator>
                 void operator()(TLCommunicator& tlcomm) const
                 {
-                    in_node1_full(tlcomm);
+                    if (in_node1())
+                        rank_barrier(tlcomm);
```

Comment:
Not sure here: shouldn't the other threads also progress until the main thread (which does the mpi barrier) indicates that they can stop progressing?

Reply:
@boeschf right. All while loops should call the progress function..

Reply:
@angainor when polling in the thread-barrier ucx gave me errors (access violation on very small addresses... not sure why). I assumed the thread barrier was fast enough. I can try when the tests pass reliably. Now they don't.